### PR TITLE
docs: add docs on commit squashing; reorganize

### PR DESCRIPTION
**Pull request by NWRichmond**
### WHY are these changes introduced?

Addresses [LS-1198](https://nacelle.atlassian.net/browse/LS-1198)

> Add instructions / tips for squashing commits into conventional commits in `nacelle-js` CONTRIBUTING.md

### WHAT is this pull request doing?

- Sets expectations & adds resources for commit squashing in `CONTRIBUTING.md`.
- Moves local development setup & new project scaffolding from `README.md` to `CONTRIBUTING.md`.
- Adds npm package version badges (with links to the package page on npm) to the `README.md`s of:
  - [`packages/shopify-checkout`](https://github.com/getnacelle/nacelle-js/tree/docs/LS-1198-document-commit-squashing-for-contributors/packages/shopify-checkout#nacelleshopify-checkout)
  - [`packages/vue`](https://github.com/getnacelle/nacelle-js/tree/docs/LS-1198-document-commit-squashing-for-contributors/packages/vue#nacellevue)
- Adds some usage instructions to `README.md` to explain how we expect merchant developers to interact with `packages/`, `examples/`, and `starters/`. This should help prevent merchant developers from unnecessarily cloning `nacelle-js`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,33 +24,98 @@ Before contributing we ask that you review the following guidelines. If you thin
...
 ## Code Of Conduct
 
 This project and everyone participating in it is governed by the [Nacelle Code of Conduct](CODE_OF_CONDUCT.md). All participants are expected to abide by this code. Please report unacceptable behavior to [support@getnacelle.com](mailto:support@getnacelle.com).
 
 ## Getting Started
 
-Nacelle-js is a set of tools designed to empower developers to build unique, performant & scalable storefronts. The repository is broken the following parts.
+### Setting up your development environment
+
+This project uses [Lerna](https://lerna.js.org) to coordinate dependencies throughout the monorepo. Lerna allows us to make changes to a package and see the effects of those changes in other packages and example projects, without the need to manually mangage package symlinks via `npm link`.
```
Comment (KrisQ) [inline CONTRIBUTING.md:41]: Hey Nick! I think there is a little typo on "manually manage" here

**Review by KrisQ — CHANGES_REQUESTED** [review]
Hey @NWRichmond everything looks great except for a little typo

File: CONTRIBUTING.md (hunk 1 above)
Comment (NWRichmond) [reply CONTRIBUTING.md]: LOLOL @ "mangage" - good find, @KrisQ!

Fixed in https://github.com/getnacelle/nacelle-js/pull/36/commits/9dd0afede48210412057c4e5601d93a1e7b7235f.

**Review by KrisQ — APPROVED** [review]
Looks good!

**Pull request merged**